examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,412 @@
+
+N_SUBJECTS = 5
+N_REGIONS = 6
+
+fmri_data = fetch_development_fmri(n_subjects=N_SUBJECTS)
+fmri_img = concat_imgs(fmri_data.func)
+
+output_dir = Path.cwd() / "results" / "plot_large_fmri_img"
+output_dir.mkdir(parents=True, exist_ok=True)
+print(f"Large fmri file will be saved to:\n{output_dir}")
+
+fmri_path = output_dir / "large_fmri.nii.gz"
+fmri_img.to_filename(fmri_path)
+
+# %%
+# Create a set of binary masks
+# ----------------------------
+# We will now create 6 binary masks from a brain atlas. Here we will use the
+# DIFUMO atlas via the :func:`~nilearn.datasets.fetch_atlas_difumo` function.
+# This atlas is a probabilistic atlas so we will threshold it to create binary
+# masks. You might want to use a different atlas or create your own masks.
+
+from nilearn.datasets import fetch_atlas_difumo
+from nilearn.image import (
+    index_img,
+    iter_img,
+    load_img,
+    new_img_like,
+    resample_to_img,
+)
+
+atlas_path = fetch_atlas_difumo(dimension=64).maps
+
+masks = load_img(atlas_path)
+# only keep the first 6 regions
+masks = index_img(masks, slice(0, N_REGIONS))
+
+# %%
+# In this case, the atlas and the mask have different resolutions. So we will
+# resample the mask to the fMRI image. It is important to do that because
+# only :class:`~nilearn.maskers.NiftiMasker` can handle the resampling of the
+# mask to the fMRI image but not the numpy indexing method we will use later.
+
+mask_paths = []
+mask_imgs = []
+for i, mask in enumerate(iter_img(masks)):
+    resampled_mask = resample_to_img(
+        mask,
+        fmri_path,
+        interpolation="nearest",
+        copy_header=True,
+        force_resample=True,
+    )
+    data = resampled_mask.get_fdata()
+    data[data != 0] = 1
+    resampled_mask = new_img_like(
+        ref_niimg=resampled_mask,
+        data=data,
+        affine=resampled_mask.affine,
+        copy_header=True,
+    )
+    mask_imgs.append(resampled_mask)
+
+    path = output_dir / f"mask_{i}.nii.gz"
+    resampled_mask.to_filename(path)
+    mask_paths.append(path)
+
+# %%
+# Mask the fMRI image using NiftiMasker
+# -------------------------------------
+# Let's first see how we would typically mask the fMRI image using
+# :class:`~nilearn.maskers.NiftiMasker`. This is the most common way to extract
+# data from an fMRI image as it makes it easy to standardize, smooth, detrend,
+# etc. the data.
+#
+# We will define a function that will do that so that it's easier to use it
+# with the ``memory_profiler`` package to measure the memory usage.
+
+from nilearn.maskers import NiftiMasker
+
+
+def nifti_masker_single(fmri_path, mask_path):
+    return NiftiMasker(mask_img=mask_path).fit_transform(fmri_path)
+
+
+# %%
+# Furthermore, we can input the fmri image and the masks in two different ways:
+#
+# 1. Using the file paths
+# 2. Using the in-memory objects
+#
+# So we will measure the memory usage for both cases.
+#
+# Let's first create a dictionary to store the memory usage for each method
```

Comment:
I don't think I would introduce this here ; as discussed, it really should only impact the method (2) using array proxy with `nibabel` and `joblib`

Reply:
So now that I've fixed my silly mistake. We do indeed see the difference between the two cases (NiftiMasker path vs. NiftiMasker in-memory). 

I don't know why that is the case, but as long as it exists, I think we should be highlighting that here.

We could probably look into the why question if we decide to improve things with NiftiMasker in future.

Reply:
Given the current pattern of results, let's try to explain this difference in more detail somewhere and then (at least) point to that explanation (if it's not directly in this text).